conans/util/files.py

```diff
@@ -187,10 +187,16 @@ def save(path, content, only_if_modified=False, encoding="utf-8"):
         only_if_modified: file won't be modified if the content hasn't changed
         encoding: target file text encoding
     """
-    try:
-        os.makedirs(os.path.dirname(path))
-    except Exception:
-        pass
+    dir_path = os.path.dirname(path)
+    if not os.path.isdir(dir_path):
+        errmsg = "The folder {} does not exist and could not be created ({})."
+        try:
+            os.makedirs(os.path.dirname(path))
+        except OSError as error:
+            if "permission denied" in error.strerror.lower():
```

Comment:
I don't like it, but it's required because of test_get_gunzip. Maybe we should use errno.EACCES instead

Reply:
This indeed reads ugly and shouldn't be necessary. Maybe the test should be reviewed too.
It should be simpler, we already checked that the folder didn't exist. So, except some weird concurrency issue, this makedirs should succeed and create the folder. Probably just checking errno.EEXIST, just to prevent any concurrency issue, and any other makedir exception should just be reraised (with an improved error message if necessary)

Reply:
Sounds a good solution

Reply:
The test `test_get_gunzip` raises `errno.ENOENT`, or "No such file or directory"